torch/_prims/utils.py

```diff
@@ -1115,6 +1258,35 @@ def make_contiguous_strides_for(shape: ShapeType) -> Tuple[int, ...]:
     return result
 
 
+def make_channels_last_2d_strides_for(shape: ShapeType) -> Tuple[int, ...]:
+    # TODO: maybe inform the user of channels_last_3d if rank of the tensor is 5?
+    check(
+        len(shape) == 4,
+        lambda: "Only tensors of rank 4 can use the channels_last memory format",
+    )
+
+    shape = list(shape)
+    shape[1], shape[-1] = shape[-1], shape[1]
```

Comment:
this is not correct, you need to permute shape elements like 
```
shape[1], shape[2], shape[3] = shape[2], shape[3], shape[1]
```
 (and how are tests passing? even for (2,2,2,2) they shouldn't. Oh, I guess it's because tests actually don't test strides)